### PR TITLE
Add types for `cds.cli`

### PR DESCRIPTION
**Pull request by chgeo**
See https://cap.cloud.sap/docs/node.js/cds-facade#cds-cli

### CODE REVIEW

File: apis/facade.d.ts [hunk 1]
```diff
@@ -29,5 +29,16 @@ export const version: string
 export const home: string
 export const root: string
 
+/**
+ * The parsed effective `cds` CLI command and arguments.
+ * May be undefined if not started from the `cds` CLI.
+ * @see https://cap.cloud.sap/docs/node.js/cds-facade#cds-cli
+ */
+export const cli: {
+  /** Basic command like `serve` */ command?: string,
```
Comment (daogrady) [inline apis/facade.d.ts:38]: do we have a full list of possible commands? I.e. `'serve' | ...` Or does this list change too much for us to maintain it with reasonable effort?

File: apis/facade.d.ts [hunk 2]
```diff
@@ -29,5 +29,16 @@ export const version: string
...
+  /** Positional arguments */       argv?: string[],
+  /** Named arguments */            opts?: Record<string, any>,
```
Comment (swaldmann) [inline apis/facade.d.ts:40]:
```suggestion
  /** Named arguments */            options?: Record<string, any>,
```

Comment (chgeo) [reply apis/facade.d.ts:40]: Ah you are right.

File: apis/facade.d.ts (hunk 1 above)
Comment (chgeo) [reply apis/facade.d.ts:38]: It's basically the full list of commands: 

add          build        compile      deploy       eval         gen          import       lint         logout       mock         pull         repl         serve        test         upgrade      watch                                         
bind         cds-tsx      completion   env          extend       help         init         login        migrate      parse        push         run          subscribe    unsubscribe  version

So quite a list.  Would not want to hard wire this.

Comment (swaldmann) [reply apis/facade.d.ts:38]: Could we add a list for Intellisense, but keep `string` as a fallback?

Comment (chgeo) [reply apis/facade.d.ts:38]: Sounds like a good compromise - a list of most used commands plus any other string.

Comment (daogrady) [reply apis/facade.d.ts]: While I love the idea, I think it [doesn't work](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAwg9gWwQQwHYBMDOUC8UDkmEATgG4T5QA+BA7ssAMYAW+AUGwPSdS1zEBrbAHNiEBlABm-KHHLEANnGRY2kgK6pGwAJZxUUCAA8IjdcGgAKMAC5YiFBkwBKO2hABuNZu16Dx03MrWyhMYGIdVGFXKHcvDS1dfUMTMwsoazt4JDQsalDwyOioAG8AXw4AtIhLfHxnKChuAuRiYChQMCKoACJaHqhmEmgAWgA+WPM4KEZEMAUICw5Z1DCZhR07ErZGxqMsh1zsGjCIqLYy3FKd3f2CSiaeMNb2zu6+gaGxKHGoVGnkFMZnMFulLPoFCAOkNesgAEaMAb6CDOC5AA) for the declaration of types for constants, only for function overloads.
Feel free to prove me wrong though! :)

Comment (daogrady) [reply apis/facade.d.ts]: duplicating keys in an object [errors](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAwgNgSxgewLaoIYDsAmBnKAXigHIMccSoAfUgIwFcE5KbSBjNMZiK2knBDBxkIPqQSowyAE7BxJBFgTy2JPBBkA3XmwAUeYDKUBzKADIoAbwC+ASgBQAegBULh1BdQAKgAtoYBgyGjhQEABm4RDswAg6UAAG7PgJsAAyAJJQnOjYoXlQQSYMqBBYwHgAdB5eALIYIFB00Ay4EUoQoQjhUFjIwFCGQcCdUOEyaFDA-onJeKkwmdWeUAACGtC+wMBgeABcTk7sGGCV7CIMOJV4J044yOx4Tn2ClQBWT3MAtOEYx4IAYm+5wQNScDggAA9pHJssgsIZsog9tYPFBXF4AEIYPAIdhw3K4KCIADW0ASGm0EFSLicBMwuAA-CjDMYsCYADRojFQbG4-E5BmhUnkyk6Gl0wV5ZmwRAoQn4LlQdFuKAABWQuNi8IwcEKMmKpXKBFpyrNyqKWhlrNMAG0ALpKlVeAByGFK+QNJTKFU8dPNAeVyDA2oRMoAStFZDgADw29kcwpYEAAPi5NjYrUE4Q6oWIVjRyqluBRJBIDhsDiAA) on TSPlayground for me 🤔

Comment (daogrady) [reply apis/facade.d.ts]: ah, it helps to not copy the old code ;)

**Review by daogrady — APPROVED** [review]

**Review by swaldmann — APPROVED** [review]

**Pull request merged**